tests/delete_regress/tests.py

```diff
@@ -408,9 +410,21 @@ def test_set_querycount(self):
         Item.objects.create(
             version=version,
             location=location,
-            location_default=location,
             location_value=location,
         )
-        # 3 UPDATEs for SET of item values and one for DELETE locations.
-        with self.assertNumQueries(4):
+        # 2 UPDATEs for SET of item values and one for DELETE locations.
+        with self.assertNumQueries(3):
             location.delete()
+
+
+# Issue #35073
+class SetCallableTests(TestCase):
+    def test_set(self):
+        Owner.objects.create(name="a")
+        Owner.objects.all().delete()
+        self.assertEqual(list(Owner.objects.all()), [])
+
+    def test_set_default(self):
+        Owner2.objects.create(name="a")
+        Owner2.objects.all().delete()
+        self.assertEqual(list(Owner2.objects.all()), [])
```

Comment:
It's not clear what we're testing here, I'd use a single `Toy` model and add a comment, e.g.
```python
class SetCallableCollectorDefaultTests(TestCase):
    def test_set(self):
        # Collector doesn't call callables used by models.SET and
        # models.SET_DEFAULT if not necessary.
        Toy.objects.create(name="test")
        Toy.objects.all().delete()
        self.assertSequenceEqual(Toy.objects.all(), [])
```

Reply:
Updated